mutation-testing/pr-differences/action.yml

```diff
@@ -0,0 +1,173 @@
+          else
+            echo "$regex_pattern" > mutants_by_packages/regex_small.txt
+          fi
+        done
+
+        exit 0
+      working-directory: actions-repo/mutation-testing/shell-scripts
+
+    - name: Run mutants
+      shell: bash
+      run: |
+        # Disable immediate exit on error
+        set +e
+
+        # Check which type of mutants to run: big, small, with or without shards, then capture the exit code
+        if [[ ${{ inputs.shard }} == -1 ]]; then
+          if [[ ${{ inputs.package-dimension }} == big ]]; then
+            if [[ -f mutants_by_packages/regex_big.txt ]]; then
+              echo "Running mutants without shards on big packages"
+              cargo mutants --no-shuffle -vV -F "$(<mutants_by_packages/regex_big.txt)" --output ./ -- --all-targets -- --test-threads 1
+              exit_code=$?
+            else
+              echo "File containing mutants doesn't exist!"
+            fi
+          else
+            if [[ -f mutants_by_packages/regex_small.txt ]]; then
+              echo "Running mutants without shards on small packages"
+              cargo mutants --no-shuffle -vV -F "$(<mutants_by_packages/regex_small.txt)" --output ./ -- --all-targets
+              exit_code=$?
+            else
+              echo "File containing mutants doesn't exist!"
+            fi
+          fi
+        else
+          if [[ ${{ inputs.package-dimension }} == big ]]; then
+            if [[ -f mutants_by_packages/regex_big.txt ]]; then
+              echo "Running mutants with shard ${{ inputs.shard }} on big packages"
+              cargo mutants --no-shuffle -vV -F "$(<mutants_by_packages/regex_big.txt)" --shard ${{ inputs.shard }}/8 --output ./ -- --all-targets -- --test-threads 1
+              exit_code=$?
+            else
+              echo "File containing mutants doesn't exist!"
+            fi
+          else
+            if [[ -f mutants_by_packages/regex_small.txt ]]; then
+              echo "Running mutants with shard ${{ inputs.shard }} on small packages"
+              cargo mutants --no-shuffle -vV -F "$(<mutants_by_packages/regex_small.txt)" --shard ${{ inputs.shard }}/4 --output ./ -- --all-targets
+              exit_code=$?
+            else
+              echo "File containing mutants doesn't exist!"
+            fi
+          fi
+        fi
+
+        # Create the folder only containing the outcomes (.txt files) and make a file containing the exit code of the command
+        mkdir mutants-shard-${{ inputs.package-dimension }}-${{ inputs.shard }}
+        echo "$exit_code" > ./mutants-shard-${{ inputs.package-dimension }}-${{ inputs.shard }}/exit_code.txt
+        mv ./mutants.out/*.txt mutants-shard-${{ inputs.package-dimension }}-${{ inputs.shard }}/
+
+        # Enable immediate exit on error again
+        set -e
+      working-directory: actions-repo/mutation-testing/shell-scripts
+
+    - name: Upload artifact
+      uses: actions/upload-artifact@v3
```

Comment:
```suggestion
      uses: actions/upload-artifact@1eb3cb2b3e0f29609092a73eb033bb759a334595 # v4.1.0
```

Reply:
Did this change as well on the new PR https://github.com/ASuciuX/stacks-core/pull/47. It is more for testing and making sure everything works as expected without bloating to much the commits here.